### PR TITLE
[JIT] Bugfixes and JSON parsing

### LINKED ISSUE #338: [JIT] JIT compiler does not support the str.parse method
This is the only way to cast string values to other types in sql.
`UnknownFunction { expr_id: v6, function: "dbsp.str.parse" }'`

### PR DESCRIPTION
**Pull request by Kixiron**
Fixes #338 and #400, implements basic json deserialization for the jit

### CODE REVIEW

**Comment by Kixiron** [comment]
The CI error comes from `python-bindings-checker` so I'm not sure how that happened

File: crates/adapters/src/catalog.rs [hunk 1]
```diff
@@ -68,8 +68,9 @@ pub struct OutputCollectionHandles {
     /// [`Stream::neighborhood`] operator.
     pub neighborhood_handle: Box<dyn SerOutputBatchHandle>,
 
-    /// A stream that contains the full snapshot of the neighborhood.  Only produces
-    /// an output whenever the `neighborhood_descr_handle` input is set to `Some(..)`.
+    /// A stream that contains the full snapshot of the neighborhood.  Only
```
[the remark this answers is not in the document]
Comment (Kixiron) [reply crates/adapters/src/catalog.rs:71]: I ran `cargo fmt`, I'm unsure as to why formatting isn't being enforced within CI

File: crates/dataflow-jit/src/codegen/json.rs [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+                unsafe {
+                    deserialize_json(
+                        uninit.as_mut_ptr(),
+                        &json_value as *const serde_json::Value as *const u8,
+                    );
+                }
+
+                let row = unsafe { uninit.assume_init() };
+                println!(
+                    "input json: {json:?}\nrow value for {}: {row:?}",
```
[the remark this answers is not in the document]
Comment (Kixiron) [reply crates/dataflow-jit/src/codegen/json.rs:256]: Yes, these were my debugging tests. Proper ones are in progress

File: crates/adapters/src/catalog.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryzhyk) [reply crates/adapters/src/catalog.rs:71]: In any case, you can run `rustfmt` on your crate only to avoid this churn.

**Comment by mihaibudiu** [comment]
I found two regressions, both fail in similar ways: `thread 'test19' panicked at 'internal error: entered unreachable code: cast from bool to i64 in v11', /home/mbudiu/git/dbsp/crates/dataflow-jit/src/codegen/mod.rs:2180:23` and `cast from i32 to decimal in v5', /home/mbudiu/git/dbsp/crates/dataflow-jit/src/codegen/mod.rs:2180:23`

**Pull request closed without merging**